Structure_Engine/Create/Physical/ISurface.cs

```
@@ -0,0 +1,84 @@
+        /***************************************************/
+        [Description("Creates a physical surface element from a Panel. The Floor will be assigned a Construction based on the SurfaceProperty of the Panel, and the PanelType of the SufaceProperty will determine what type of surface element to create, unless overridden.")]
+        [Input("panel", "The Panel to use as the base for the framing element.")]
+        [Input("structuralUsage", "The type of surface element to create. if Undefined, the type will be based on the panel's SurfaceProperty.")]
+        [Output("surfaceElement", "The created surface element based on the Panel element provided.")]
+        public static BHPE.ISurface ISurface(Panel panel, StructuralUsage2D structuralUsage = StructuralUsage2D.Undefined)
+        {
+            if (panel.IsNull())
+                return null;
+            if (structuralUsage == StructuralUsage2D.Undefined)
+            {
+                object result = panel.Property?.PropertyValue("PanelType");
+                if (result != null )
+                {
+                    structuralUsage = (StructuralUsage2D)result;
+                }
+            }
+
+            switch (structuralUsage)
+            {
+                case StructuralUsage2D.Wall:
+                    return Wall(panel);
+                case StructuralUsage2D.DropPanel:
+                case StructuralUsage2D.PileCap:
+                case StructuralUsage2D.Slab:
+                    return Floor(panel);
+                case StructuralUsage2D.Undefined:
+                default:
+                    {
+                        if (Math.Abs(panel.Normal().DotProduct(Vector.ZAxis)) == 1)
+                            return Floor(panel);
+                        else if (panel.Normal().DotProduct(Vector.ZAxis) == 0)
+                            return Wall(panel);
+                        else 
+                            return null;
```

I would throw an error that some panels could not be identified because not horizontal/vertical, so the user is requested to enforce the type

---

agreed